src/main/java/gregtech/api/recipes/ingredients/GTRecipeFluidInput.java

```diff
@@ -0,0 +1,109 @@
+package gregtech.api.recipes.ingredients;
+
+import gregtech.api.recipes.ingredients.NBTMatching.NBTMatcher;
+import gregtech.api.recipes.ingredients.NBTMatching.NBTcondition;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nullable;
+import java.lang.ref.WeakReference;
+
+public class GTRecipeFluidInput implements IGTRecipeInput {
+
+    public int amount;
+    boolean isConsumable = true;
+    NBTMatcher nbtMatcher;
+    NBTcondition nbtCondition;
+    FluidStack inputStacks;
+
+    public GTRecipeFluidInput(FluidStack inputStacks) {
+        this.inputStacks = inputStacks;
+        this.amount = inputStacks.amount;
+    }
+
+    public GTRecipeFluidInput(FluidStack inputStacks, int amount) {
+        this.inputStacks = inputStacks;
+        this.amount = amount;
+    }
+
+    public static GTRecipeFluidInput getOrCreate(FluidStack fluidStack, int amount) {
+        fluidStack.amount = amount;
+        return getFromCache(new GTRecipeFluidInput(fluidStack));
```

Comment:
Should this just call `new GTRecipeFluidInput(FluidStack, int)` instead of setting the amount of the passed fluidstack and then calling the fluidstack only method?